### PR TITLE
Make static buffers in numToString thread local

### PR DESCRIPTION
**Pull request by tlively**
Validation is performed on multiple threads at once and when there are multiple
validation failures, those threads can all end up in `numToString` at the same
time as they construct their respective error messages. Previously the threads
would race on their access to the snprintf buffers, sometimes leading to
segfaults. Fix the data races by making the buffers thread local.

### CODE REVIEW

**Review by kripken — COMMENTED** [review]
Good find!

Is it worth doing `thread_local` here? How about just removing the `static`, which would make these normal stack allocations? That seems simpler and it might actually be faster.

**Comment by tlively** [comment]
Sure, I guess a couple kilobytes on the stack shouldn't matter.

File: src/emscripten-optimizer/simple_ast.h [hunk 1]
```diff
@@ -947,17 +947,18 @@ struct JSPrinter {
     bool integer = wasm::isInteger(d);
 #define BUFFERSIZE 1000
     // f is normal, e is scientific for float, x for integer
-    static char full_storage_f[BUFFERSIZE], full_storage_e[BUFFERSIZE];
+    char full_storage_f[BUFFERSIZE] = {};
+    char full_storage_e[BUFFERSIZE] = {};
```
Comment (kripken) [inline src/emscripten-optimizer/simple_ast.h:951]: does `= {}` initialize it to 0? I don't think we need that here.

Comment (tlively) [reply src/emscripten-optimizer/simple_ast.h]: Removed.

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
Oh, it looks like the `thread_local` on the buffers might be important after all, since this function returns pointers to the buffers and the buffers need to live longer than the function call.

**Comment by kripken** [comment]
Oh, good point, yeah... Indeed that is necessary. lgtm with that.

**Pull request merged**